### ISSUE TITLE
Preference reset

### ISSUE THREAD

**Issue by mrvkino** [laconalabs/LaconaApp#142]
The keyboard shortcut to call lacona and the command aliases are always resetted to the default when lacona starts.

**Comment by sebj** [comment]
+1 – literally all preferences are reset, none are saved (e.g. toggle launch at login off, quit, reopen and it's on).

**Comment by brandonhorst** [comment]
This is very interesting - I have never seen this or heard it reported. I'm going to dive in and try to reproduce it immediately.

**Comment by brandonhorst** [comment]
Could someone experiencing the problem please

1. Make a change to the Lacona settings (Keyboard Shortcut, for example)
2. Quit Lacona
3. Send me the output of this Terminal command:

```
defaults read io.lacona.Lacona
```

You can post it here or email it to [app@lacona.io](mailto:app@lacona.io).

**Comment by mrvkino** [comment]
It started with the new version with the big re-write of the preference pane.

```
{
    LaconaContactCalendarAccessPrompted = 1;
    LaconaTutorialPopoverShown = 1;
    "NSToolbar Configuration 1D60155D-F445-4AF2-892E-62BBC693D001" =     {
        "TB Display Mode" = 1;
        "TB Icon Size Mode" = 1;
        "TB Is Shown" = 1;
        "TB Size Mode" = 1;
    };
    "NSWindow Frame SUUpdateAlert" = "410 363 620 392 0 0 1440 877 ";
    SUAutomaticallyUpdate = 1;
    SUEnableAutomaticChecks = 1;
    SUHasLaunchedBefore = 1;
    SULastCheckTime = "2016-11-23 16:33:50 +0000";
    SUSendProfileInfo = 0;
    addonConfig = "{\\"lacona-convert-currency\\":{\\"defaultCurrency\\":\\"CAD\\"},\\"currencyConversion\\":{\\"defaultCurrency\\":\\"CAD\\"}}";
    addons = "[{\\"name\\":\\"lacona-applescript\\",\\"enabled\\":true},{\\"name\\":\\"lacona-timer\\",\\"enabled\\":true}]";
    generalConfig = "{\\"_general\\":{\\"keyboardShortcut\\":{\\"string\\":\\" \\",\\"display\\":\\"Space\\",\\"alt\\":false,\\"ctrl\\":true,\\"key\\":49,\\"cmd\\":false,\\"shift\\":false}},\\"_shortcuts\\":{\\"aliases\\":[{\\"replace\\":\\"o\\",\\"with\\":\\"open\\"},{\\"replace\\":\\"rmt\\",\\"with\\":\\"remind me to\\"},{\\"replace\\":\\"next\\",\\"with\\":\\"skip\\"}]}}";
}
```

**Comment by brandonhorst** [comment]
Fastinating. And after you start Lacona up again, the Keyboard Shortcut switches back to Option+Space? Does your custom Command Alias stick around?

**Comment by mrvkino** [comment]
No, they all disappear. Here is the output after starting lacona.

```
⋊> defaults read io.lacona.Lacona
{
    LaconaContactCalendarAccessPrompted = 1;
    LaconaTutorialPopoverShown = 1;
    "NSToolbar Configuration 1D60155D-F445-4AF2-892E-62BBC693D001" =     {
        "TB Display Mode" = 1;
        "TB Icon Size Mode" = 1;
        "TB Is Shown" = 1;
        "TB Size Mode" = 1;
    };
    "NSWindow Frame SUUpdateAlert" = "410 363 620 392 0 0 1440 877 ";
    SUAutomaticallyUpdate = 1;
    SUEnableAutomaticChecks = 1;
    SUHasLaunchedBefore = 1;
    SULastCheckTime = "2016-11-23 16:42:08 +0000";
    SUSendProfileInfo = 0;
    addonConfig = "{\\"lacona-convert-currency\\":{\\"defaultCurrency\\":\\"CAD\\"},\\"currencyConversion\\":{\\"defaultCurrency\\":\\"CAD\\"}}";
    addons = "[{\\"name\\":\\"lacona-applescript\\",\\"enabled\\":true},{\\"name\\":\\"lacona-timer\\",\\"enabled\\":true}]";
    generalConfig = "{}";
}
```

**Comment by sebj** [comment]
Same here. It looks like preferences are only reset when I next open Lacona. If I quit, they're still all saved.

**Comment by brandonhorst** [comment]
Thanks for the info. Still no idea what is causing this but I should have enough to go on.

**Comment by brandonhorst** [comment]
Ah, got it. Thank you so much for the info. Fixing it right away.

You can fix the problem immediately by running

```
defaults delete io.lacona.Lacona addonConfig
```

You will need to reset the Convert Currency settings and any other Addon settings you've configured, so if that's an issue just wait for the fix.

**Comment by sebj** [comment]
That looks like it's fixed it for me (changed settings, quit, reopened a few times - all good so far). Thanks for the quick response!

**Comment by mrvkino** [comment]
Seems to work.

Thanks.

**Comment by brandonhorst** [comment]
This should be fixed by 1.1.2 that I just pushed out! Sorry about the inconvenience!
